exercises/CI/introduction_to_ci/README.md

```diff
@@ -0,0 +1,154 @@
+
+### Continuous Integration Servers
+Continuous integration servers are essentially glorified schedulers. They sit looking at your Version Control System (VCS) and when they see that a change has been introduced they do what ever they have been configured to do. E.g. to compile the new version of the code base and run a set of tests to see if it still works. Different CI servers may have different user interfaces, however they essentially all do the same thing and have a very common feature set. Therefore learn how to use one CI server and you should find that switching to another is a pretty easy process.
+
+## Exercise Learning Objectives
+- Introduction Concourse CI
+
+## Exercise
+**Note:** Before going any further do the following:
+- `cd YOUR_CLONE_OF_THIS REPO`
+- `source ./bin/env`
+- `cd `
+
+In this exercise you will be working using a CI server called Concourse. We've done the hard bit and have provided you with your very own Concourse installation. We've also provided you with a GIT server for your CI server to talk to. To start everything up, run: cic up 
+
+If all is well you should see the following:
+```
+  [OK] Environment cic'd up :)
+
+```
+
+Click the login button at the top right of the [concourse home page](http://127.0.0.1:8080) and login with username `test` and password `test`. You'll note that there arn't currently any 'pipelines' configured. A pipeline is the name that Concourse gives to one or more tasks/jobs that have been configured to be run in a prescribed order when a particular event happens. E.g. a change is detected in GIT. Pipelines can be can be as simple as a one step process or they can be used to model very complex processes. In a later exercise we'll go in to pipelines and there features in more depth.
+
+We have supplied a very simple pipeline. Let's put it in to concourse.
+
+Some CI's have extensive UIs through which they can be configured, Concourse is primarly configured via it's commandline interface (CLI) which makes webservice calls to configure the Concourse server. The CLIs for each O/S are available in `./resources` directory. Use the fly binary that is correct for your O/S to complete this exercise.
+
+First we must login, run `./resources/linux/fly -t local login --concourse-url 127.0.0.1:8080 -u test -p test`
+you should see the following output:
+```
+logging in to team 'main'
+
+
+target saved
+```
+
+Now let's push a pipeline to concourse by running: `./resources/linux/fly  -t local set-pipeline -p ci-intro-pipeline -c ./resources/pipeline.yml -n`
```

Comment:
Before we push the pipeline, we probably need a bit where the user takes a look at the pipeline file, and we explain a little about what this pipeline does, and how concourse uses 'resources' and 'tasks' to build it's pipelines automatically.